pkg/compat/metrics.go

```diff
@@ -0,0 +1,126 @@
+package compat
+
+import (
+	ref "github.com/konveyor/mig-controller/pkg/reference"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	api "k8s.io/apimachinery/pkg/runtime"
+	"runtime"
+	"strings"
+)
+
+//
+// Labels.
+const (
+	Get       = "Get"
+	List      = "List"
+	Create    = "Create"
+	Update    = "Update"
+	Delete    = "Delete"
+	Cluster   = "cluster"
+	Component = "component"
+	Function  = "function"
+	Kind      = "kind"
+	Method    = "method"
+)
+
+//
+// Global reporter.
+var Metrics *Reporter
+
+func init() {
+	Metrics = &Reporter{
+		counter: promauto.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "mtc_client",
+				Help: "MTC client API metrics.",
+			},
+			[]string{
+				Cluster,
+				Component,
+				Function,
+				Kind,
+				Method,
+			}),
+	}
+}
+
+//
+// Metric reporter.
+type Reporter struct {
+	counter *prometheus.CounterVec
+}
+
+//
+// Report `get` API call.
+func (m *Reporter) Get(client Client, object api.Object) {
+	m.report(client, Get, object)
+}
+
+//
+// Report `list` API call.
+func (m *Reporter) List(client Client, object api.Object) {
+	m.report(client, List, object)
+}
+
+//
+// Report `create` API call.
+func (m *Reporter) Create(client Client, object api.Object) {
+	m.report(client, Create, object)
+}
+
+//
+// Report `update` API call.
+func (m *Reporter) Update(client Client, object api.Object) {
+	m.report(client, Update, object)
+}
+
+//
+// Report `delete` API call.
+func (m *Reporter) Delete(client Client, object api.Object) {
+	m.report(client, Delete, object)
+}
+
+//
+// Determine the call context.
+func (m *Reporter) context() (component, function string) {
```

Comment:
What's a practical example of this `context()`? E.g. what will the return values be.

Reply:
> What's a practical example of this `context()`? E.g. what will the return values be.

The component (which controller) and the package qualified function name making the API call.